packages/@lwc/babel-plugin-component/src/decorators/wire/validate.ts

```diff
@@ -26,65 +25,71 @@ function validateWireId(id: NodePath | undefined, path: NodePath, state: LwcBabe
         );
     }
 
-    const isMemberExpression = id.isMemberExpression();
+    if (id.isIdentifier()) {
+        const binding = path.scope.getBinding(id.node.name);
 
-    if (!id.isIdentifier() && !isMemberExpression) {
-        throw generateError(
-            id,
-            {
-                errorInfo: DecoratorErrors.FUNCTION_IDENTIFIER_SHOULD_BE_FIRST_PARAMETER,
-            },
-            state
-        );
-    }
+        if (!binding) {
+            throw generateError(
+                id,
+                {
+                    errorInfo: DecoratorErrors.WIRE_ADAPTER_SHOULD_BE_IMPORTED,
+                    messageArgs: [id.node.name],
+                },
+                state
+            );
+        }
 
-    if (id.isMemberExpression({ computed: true })) {
-        throw generateError(
-            id,
-            {
-                errorInfo: DecoratorErrors.FUNCTION_IDENTIFIER_CANNOT_HAVE_COMPUTED_PROPS,
-            },
-            state
-        );
-    }
+        if (!binding.path.isImportSpecifier() && !binding.path.isImportDefaultSpecifier()) {
+            throw generateError(
+                id,
+                {
+                    errorInfo:
+                        DecoratorErrors.IMPORTED_FUNCTION_IDENTIFIER_SHOULD_BE_FIRST_PARAMETER,
+                },
+                state
+            );
+        }
+    } else if (id.isMemberExpression()) {
+        if (id.node.computed) {
```

Comment:
These refactorings seem pretty extensive, and I'm concerned that our boolean logic is not 1:1 the same because our tests may not be extensive enough here.

Do we need to refactor this so aggressively? can we keep the same boolean logic as before?

Reply:
Ditto for the code below in the same file.

Reply:
Makes sense. I'll try to minimize these changes.